### PR TITLE
[INFRA-1196] Add tests for update-site redirects

### PR DESCRIPTION
**Pull request by olivergondza**
@rtyler, It is incomplete and new tests can be added but it is enough to reproduce the problem.

### CODE REVIEW

**Comment by olivergondza** [comment]
It seems that PR building is not configured so you will have to either trust me it works or grant me more permissions on ci.jenkins.io ...

**Comment by oleg-nenashev** [comment]
Why would you need to remove readme and LICENSE?

**Comment by olivergondza** [comment]
Dammit, it should not have been filled against master - it needs to go to `update-site` branch. GitHub does not seem to permit PRs to branches that does not exist and I do not have the commit access to push directly.

Thanks for spotting it, Oleg. That is the reason why it seems to remove the files.

@rtyler

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+properties([
+    [$class: 'jenkins.model.BuildDiscarderProperty', strategy: [$class: 'LogRotator', numToKeepStr: '5']],
+    pipelineTriggers([cron('@hourly')]),
+])
+
+node('docker') {
+    timestamps {
+        docker.image('debian').inside('-u 0:0') {
+            stage('Prepare Container') {
+                sh 'apt-get update -qy && apt-get install -qy --force-yes curl'
```
Comment (jglick) [inline Jenkinsfile]: Better to start with a container that already has these tools installed, like `cloudbees/java-build-tools` off the top of my head.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            }
+
+            stage('Run the tests') {
+                git url: '/home/ogondza/code/jenkins/infra-acceptance-tests', branch: "update-site"
```
Comment (jglick) [inline Jenkinsfile]: Uh, how is this going to work?

File: version-redirects.test.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Test Jenkins requests for metadata gets redirected to correct version of update center.
+
+UPDATE_CENTER='https://updates.jenkins-ci.org/'
+
+result=0
+
+function checkRedirect() {
+  version="$1"
+  target="$2"
+  stderr="$(curl -vL "${UPDATE_CENTER}/update-center.json?id=default&version=${version}" 2>&1 > /dev/null)"
```
Comment (jglick) [inline version-redirects.test.sh]: Did you want to pass `-I`?

**Review by jglick — COMMENTED** [review]
A test like this will notify us of problems, but only after the problematic site is live. I think we should explore making the build for `backend-update-center2` itself perform checks like this, so bad configurations never get deployed. Would I suppose mean running a test container with Apache and sending sample requests to it.

**Comment by jglick** [comment]
[INFRA-1196](https://issues.jenkins-ci.org/browse/INFRA-1196) FTR

**Comment by olivergondza** [comment]
> A test like this will notify us of problems, but only after the problematic site is live. I think we should explore making the build for backend-update-center2 itself perform checks like this, so bad configurations never get deployed. Would I suppose mean running a test container with Apache and sending sample requests to it.

That would be better, no question about it. I lean toward getting this meged in the meantime ...

File: Jenkinsfile (hunk 2 above)
Comment (olivergondza) [reply Jenkinsfile]: Works on my machine...

**Comment by olivergondza** [comment]
@rtyler, can we move on with this?

**Pull request merged**